### PR TITLE
Jpm makefile - as described in https://github.com/lz4/lz4/issues/688

### PR DESCRIPTION
**Pull request by JPeterMugaas**
As per https://github.com/lz4/lz4/issues/688 discussion.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,15 +36,50 @@ TESTDIR = tests
...
+ifneq (,$(filter CYGWIN%,$(TARGET_OS)))
 EXT  = .exe
-VOID = nul
 else
 EXT  =
-VOID = /dev/null
+endif
+endif
+endif
 endif
```
Comment (Cyan4973) [inline Makefile]: I'm a bit lost in the nesting levels here.
Could it be indented ?

File: Makefile [hunk 2]
```diff
@@ -36,15 +36,50 @@ TESTDIR = tests
...
+TARGET_OS ?= $(shell uname)
+ifeq ($(TARGET_OS),)
+  TARGET_OS ?= $(OS)
+endif
+
+ifneq (,$(filter Windows%,$(TARGET_OS)))
+LIBLZ4 = liblz4-$(LIBVER_MAJOR)
+EXT  = .exe
+else
+EXT  = .exe
```
Comment (Cyan4973) [inline Makefile]: It seems this line will define `EXT` as `=.exe`, so why redefining the same afterwards ?

File: Makefile [hunk 3]
```diff
@@ -36,15 +36,50 @@ TESTDIR = tests
...
+ifneq (,$(filter MINGW%,$(TARGET_OS)))
```
Comment (Cyan4973) [inline Makefile]: Several of these tests have the same outcome, could they be coalesced ?

File: Makefile [hunk 4]
```diff
@@ -36,15 +36,50 @@ TESTDIR = tests
...
-VOID = nul
 else
 EXT  =
-VOID = /dev/null
+endif
+endif
+endif
 endif
 
+#determine if dev/nul based on host environment
```
Comment (Cyan4973) [inline Makefile]: We have about the same questions regarding readability and nest levels here.

**Comment by Cyan4973** [comment]
Generally speaking, I believe this patch goes in the right direction.
Outcome is more correct for Windows.
I would just try to simplify the top blocks, which define `.exe` or `nul`, to make them more readable.

File: Makefile (hunk 1 above)
Comment (JPeterMugaas) [reply Makefile]: Yes, they are now indented.  And I also tried to isolate some of tests that I had made or worked on.  They are now in a new Makefile.inc.  In fact, there are some things that probably belong in Makefile.inc related to some other platform stuff as well as some run-of-mill macros that's duplicate cruft.

File: programs/Makefile [hunk 5]
```diff
@@ -56,17 +56,48 @@ LZ4_VERSION=$(LIBVER)
 MD2ROFF   = ronn
 MD2ROFF_FLAGS = --roff --warnings --manual="User Commands" --organization="lz4 $(LZ4_VERSION)"
```
Comment (Cyan4973) [inline programs/Makefile]: Can't `programs/Makefile` use `include Makefile.inc` too ?

Comment (JPeterMugaas) [reply programs/Makefile]: It should.

**Comment by Cyan4973** [comment]
This patch looks good to me.
I can merge it now if you wish.

Or do you plan to work a bit more on it ?

**Comment by JPeterMugaas** [comment]
Go ahead.  What I would do with version info could be a separate checkin.

**Comment by JPeterMugaas** [comment]
OOPS. I didn't mean to close this.  Go ahead and merge.

**Pull request merged**